azure/scope/machinepool.go

```diff
@@ -494,6 +502,27 @@ func (m *MachinePoolScope) createMachine(ctx context.Context, machine azure.VMSS
 	return nil
 }
 
+func (m *MachinePoolScope) deleteMachine(ctx context.Context, ampm infrav1exp.AzureMachinePoolMachine) error {
+	ctx, _, done := tele.StartSpanWithLogger(ctx, "scope.MachinePoolScope.deleteMachine")
+	defer done()
+
+	machine, err := util.GetOwnerMachine(ctx, m.client, ampm.ObjectMeta)
+	if err != nil {
+		return errors.Wrapf(err, "error getting owner Machine for AzureMachinePoolMachine %s/%s", ampm.Namespace, ampm.Name)
+	}
+	if machine == nil {
```

Comment:
It's kind of up to us on how to handle this case. Do we want to try to delete the AMPM directly in the case that the MP is deleted, or just leave it to requeue?